### ISSUE TITLE
make `required_beam_type` in recipe json accepts enum name?

### ISSUE THREAD

**Issue by ZZZank** [luccaPossamai/optical#39]
I made [an addon of Create Optical](https://github.com/ZZZank/OpticalJS) that adds KubeJS recipe support for it:

![img](https://github.com/user-attachments/assets/5e622dd8-b1cd-4c38-b37d-6a28b33c9123)

and you can see the `required_beam_type` is accepting a number that represents a enum, which is not easy to understand, an enum name will definitely be better

**Comment by luccaPossamai** [comment]
Yeah you are absolutely right. I'll see about adding this in future updates.
